hystrix-contrib/hystrix-reactivesocket-event-stream/src/main/java/com/netflix/hystrix/contrib/reactivesocket/EventStream.java

```diff
@@ -0,0 +1,92 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.hystrix.contrib.reactivesocket;
+
+import com.netflix.hystrix.config.HystrixConfigurationStream;
+import com.netflix.hystrix.contrib.reactivesocket.serialize.SerialHystrixConfiguration;
+import com.netflix.hystrix.contrib.reactivesocket.serialize.SerialHystrixDashboardData;
+import com.netflix.hystrix.contrib.reactivesocket.serialize.SerialHystrixMetric;
+import com.netflix.hystrix.contrib.reactivesocket.serialize.SerialHystrixRequestEvents;
+import com.netflix.hystrix.contrib.reactivesocket.serialize.SerialHystrixUtilization;
+import com.netflix.hystrix.metric.HystrixRequestEventsStream;
+import com.netflix.hystrix.metric.sample.HystrixUtilizationStream;
+import io.reactivesocket.Payload;
+import rx.Observable;
+
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
+
+class EventStream implements Supplier<Observable<Payload>> {
+
+    private final static int CONFIGURATION_DATA_INTERVAL_IN_MS = 500;
+    private final static int UTILIZATION_DATA_INTERVAL_IN_MS = 500;
+
+    private final Observable<Payload> source;
+    private final AtomicBoolean isSourceCurrentlySubscribed = new AtomicBoolean(false);
+
+    /* package-private */EventStream(Observable<Payload> source) {
+        this.source = source
+                .doOnSubscribe(() -> isSourceCurrentlySubscribed.set(true))
+                .doOnUnsubscribe(() -> isSourceCurrentlySubscribed.set(false))
+                .share()
+                .onBackpressureDrop();
+    }
+
+    @Override
+    public Observable<Payload> get() {
+        return source;
+    }
+
+    public static EventStream getInstance(EventStreamEnum eventStreamEnum) {
+        final Observable<Payload> source;
+
+        switch (eventStreamEnum) {
+            case CONFIG_STREAM:
+                source = new HystrixConfigurationStream(CONFIGURATION_DATA_INTERVAL_IN_MS)
+                        .observe()
+                        .map(SerialHystrixConfiguration::toBytes)
+                        .map(SerialHystrixMetric::toPayload);
+                break;
+            case REQUEST_EVENT_STREAM:
+                source = HystrixRequestEventsStream.getInstance()
+                        .observe()
+                        .map(SerialHystrixRequestEvents::toBytes)
+                        .map(SerialHystrixMetric::toPayload);
+                break;
+            case UTILIZATION_STREAM:
+                source = new HystrixUtilizationStream(UTILIZATION_DATA_INTERVAL_IN_MS)
+                        .observe()
+                        .map(SerialHystrixUtilization::toBytes)
+                        .map(SerialHystrixMetric::toPayload);
+                break;
+            case GENERAL_DASHBOARD_STREAM:
+                source = HystrixDashboardStream.getInstance()
```

Comment:
Singletons for util/config stream to allow subscribers to share w/ archaius sounds like a good approach.